src/containerbuddy/ips.go

```diff
@@ -0,0 +1,253 @@
+	 * recoverable. Let's pass on the parsed interfaces and log the error
+	 * state. */
+	if interfaceIPsErr != nil && len(interfaceIPs) > 0 {
+		log.Printf("We had a problem reading information about some network "+
+			"interfaces. If everything works, it is safe to ignore this"+
+			"message. Details:\n%s\n", interfaceIPsErr)
+	}
+
+	return findIPWithSpecs(specs, interfaceIPs)
+}
+
+// findIPWithSpecs will use the given interface specification list and will
+// find the first IP in the interfaceIPs that matches a spec
+func findIPWithSpecs(specs []interfaceSpec, interfaceIPs []interfaceIP) (string, error) {
+	// Find the interface matching the name given
+	for _, spec := range specs {
+		index := 1
+		iface := ""
+		for _, iip := range interfaceIPs {
+			// Since the interfaces are ordered by name
+			// a change in interface name can safely reset the index
+			if iface != iip.Name {
+				index = 1
+				iface = iip.Name
+			} else {
+				index++
+			}
+			if spec.Match(index, iip) {
+				return iip.IPString(), nil
+			}
+		}
+	}
+
+	// Interface not found, return error
+	return "", fmt.Errorf("None of the interface specifications were able to match\nSpecifications: %s\nInterfaces IPs: %s",
+		specs, interfaceIPs)
+}
+
+type interfaceSpec struct {
+	Spec    string
+	IPv6    bool
+	Name    string
+	Network *net.IPNet
+	Index   int
+}
+
+func (spec interfaceSpec) String() string {
+	return spec.Spec
+}
+
+func (spec interfaceSpec) Match(index int, iip interfaceIP) bool {
+	// Specific Interface eth1, eth0[1], eth0:inet6, inet, inet6
+	if spec.Name == iip.Name || spec.Name == "*" {
+		// Has index and matches
+		if spec.Index > 0 {
+			return (spec.Index == index)
+		}
+		// Don't match loopback address for wildcard spec
+		if spec.Name == "*" && iip.IP.IsLoopback() {
+			return false
+		}
+		return spec.IPv6 != iip.IsIPv4()
+	}
+	// CIDR
+	if spec.Network != nil && spec.Network.Contains(iip.IP) {
+		return true
+	}
+
+	return false
+}
+
+func parseInterfaceSpecs(interfaces []string) ([]interfaceSpec, error) {
+	var errors []string
+	var specs []interfaceSpec
+	for _, iface := range interfaces {
+		spec, err := parseInterfaceSpec(iface)
+		if err != nil {
+			errors = append(errors, err.Error())
+			continue
+		}
+		specs = append(specs, spec)
+	}
+	if len(errors) > 0 {
+		err := fmt.Errorf(strings.Join(errors, "\n"))
+		println(err.Error())
+		return specs, err
+	}
+	return specs, nil
+}
+
+var (
+	ifaceSpec = regexp.MustCompile(`^(?P<Name>\w+)(?:(?:\[(?P<Index>\d+)\])|(?::(?P<Version>inet6?)))?$`)
+)
+
+func parseInterfaceSpec(spec string) (interfaceSpec, error) {
+	if spec == "inet" {
+		return interfaceSpec{IPv6: false, Name: "*"}, nil
+	}
+	if spec == "inet6" {
+		return interfaceSpec{IPv6: true, Name: "*"}, nil
+	}
+
+	if match := ifaceSpec.FindStringSubmatch(spec); match != nil {
+		name := match[1]
+		index := match[2]
+		inet := match[3]
+		if index != "" {
+			i, _ := strconv.Atoi(index)
```

Comment:
I made it 1-index for two reasons.

1. It was easier to explain in the readme. (1 is the 1st, 2 is the 2nd, etc...)
2. It made the code a little cleaner - I don't have to use -1 instead of 0 for Index - which i have to set explicitly when creating the struct.

However; the target audience understands 0-index, so I can switch it easily enough.